packages/x-charts-premium/src/ChartsRenderer/configuration.tsx

```diff
@@ -0,0 +1,482 @@
+            options: [
+              { content: localeText.chartConfigurationOptionEnd, value: 'end' },
+              {
+                content: localeText.chartConfigurationOptionExtremities,
+                value: 'extremities',
+              },
+              { content: localeText.chartConfigurationOptionMiddle, value: 'middle' },
+              { content: localeText.chartConfigurationOptionStart, value: 'start' },
+            ],
+          },
+        }
+      : {}),
+    ...(tickOptions
+      ? {
+          tickLabelPlacement: {
+            label: localeText.chartConfigurationTickLabelPlacement,
+            type: 'select',
+            default: 'middle',
+            options: [
+              { content: localeText.chartConfigurationOptionMiddle, value: 'middle' },
+              { content: localeText.chartConfigurationOptionTick, value: 'tick' },
+            ],
+          },
+        }
+      : {}),
+    seriesAxisReverse: {
+      label: localeText.chartConfigurationSeriesAxisReverse,
+      type: 'boolean',
+      default: false,
+    },
+  },
+});
+
+const getTooltipSection = (localeText: ChartsLocaleText): GridChartsConfigurationSection => ({
+  id: 'tooltip',
+  label: localeText.chartConfigurationSectionTooltip,
+  controls: {
+    tooltipPlacement: {
+      label: localeText.chartConfigurationTooltipPlacement,
+      type: 'select',
+      default: 'auto',
+      options: [
+        { content: localeText.chartConfigurationOptionAuto, value: 'auto' },
+        { content: localeText.chartConfigurationOptionTop, value: 'top' },
+        { content: localeText.chartConfigurationOptionBottom, value: 'bottom' },
+        { content: localeText.chartConfigurationOptionLeft, value: 'left' },
+        { content: localeText.chartConfigurationOptionRight, value: 'right' },
+      ],
+    },
+    tooltipTrigger: {
+      label: localeText.chartConfigurationTooltipTrigger,
+      type: 'select',
+      default: 'axis',
+      options: [
+        { content: localeText.chartConfigurationOptionNone, value: 'none' },
+        { content: localeText.chartConfigurationOptionAxis, value: 'axis' },
+        { content: localeText.chartConfigurationOptionItem, value: 'item' },
+      ],
+    },
+  },
+});
+
+const getLegendSection = (localeText: ChartsLocaleText): GridChartsConfigurationSection => ({
+  id: 'legend',
+  label: localeText.chartConfigurationSectionLegend,
+  controls: {
+    legendPosition: {
+      label: localeText.chartConfigurationLegendPosition,
+      type: 'select',
+      default: 'top',
+      options: [
+        { content: localeText.chartConfigurationOptionNone, value: 'none' },
+        { content: localeText.chartConfigurationOptionTop, value: 'top' },
+        { content: localeText.chartConfigurationOptionBottom, value: 'bottom' },
+        { content: localeText.chartConfigurationOptionLeft, value: 'left' },
+        { content: localeText.chartConfigurationOptionRight, value: 'right' },
+      ],
+    },
```

Comment:
This doesn't fully provide all the available options. The legend can be positioned in any of the 8 cardinal directions, n, ne, e, se, s, sw, w, nw.
Although not all of them are valid depending on horizontal/vertical.

I think we should either provide all the options, or remove the `horizontal/vertical` option and handle `top/bottom=horizontal`, `left/right=vertical`